storage/google_storage.py

```diff
@@ -93,4 +106,7 @@ def delete_directory(self) -> None:
 
         for blob in bucket.list_blobs(prefix=self._parsed_storage_uri.path[1:] + "/"):
             unversioned_blob = bucket.blob(blob.name)
-            unversioned_blob.delete()
+            try:
```

Comment:
See my comment, above in the `save_to_directory` code, about raising when `list_blobs` returns an iterator of 0 blobs. I think the same applies, here.